hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/RocksDatabase.java

```diff
@@ -300,6 +301,32 @@ public void batchPut(ManagedWriteBatch writeBatch, byte[] key, byte[] value)
         counter.decrementAndGet();
       }
     }
+
+    public void batchDelete(ManagedReadWriteBatch writeBatch, byte[] key)
+        throws IOException {
```

Comment:
The code duplication between the write batch and read write batch versions of these methods is somewhat unfortunate, but I'm not sure there's much we can do about it here since RocksDB did not put WriteBatch and WriteBatchWithIndex under a common class or interface.

Reply:
@errose28 I tried to use base class, but since RocksDB defined separate method and can not have any strategy. Another way is using typecast in common method, but this will add if else complexity, So IMO having separate method is better in this case.
Its usages limitation of RocksDB.